grpcbp/server_middlewares.go

```diff
@@ -0,0 +1,81 @@
+package grpcbp
+
+import (
+	"context"
+	"strings"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+	"github.com/reddit/baseplate.go/transport"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func InjectServerSpanInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		split := strings.Split(info.FullMethod, "/")
+		m := strings.ToLower(split[len(split)-1])
+		ctx, span := StartSpanFromGRPCContext(ctx, m)
+		defer func() {
+			span.FinishWithOptions(tracing.FinishOptions{
+				Ctx: ctx,
+				Err: err,
+			}.Convert())
+		}()
+		return handler(ctx, req)
+	}
+}
+
+func InjectEdgeContextInterceptor(impl ecinterface.Interface) grpc.UnaryServerInterceptor {
+	if impl == nil {
+		impl = ecinterface.Get()
+	}
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		ctx = InitializeEdgeContext(ctx, impl)
+		return handler(ctx, req)
+	}
+}
+
+func InitializeEdgeContext(ctx context.Context, impl ecinterface.Interface) context.Context {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return ctx
+	}
+	values, ok := md[transport.HeaderEdgeRequest]
+	if !ok {
```

Comment:
```suggestion
	values := md.Get(transport.HeaderEdgeRequest)
	if len(values) == 0 {
```

Reply:
We should probably provide an abstraction when accessing the gRPC metadata that takes care of retrieving the first element and just returns a bool to check if it exists to make this code easier to read.

Reply:
yes we should. this pattern is repeated so many times in the code.